src/Featurizers/Components/GlobalMinimumFrequencyEstimator.h

```diff
@@ -0,0 +1,222 @@
+// ----------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+// ----------------------------------------------------------------------
+#pragma once
+
+#include <limits>
+
+#include "TrainingOnlyEstimatorImpl.h"
+#include "FrequencyEstimator.h"
+#include "GrainFeaturizerImpl.h"
+
+namespace Microsoft {
+namespace Featurizer {
+namespace Featurizers {
+namespace Components {
+
+// TODO: This should be removed in favor of the StatisticalMetricsEstimator
+
+static constexpr char const * const         GlobalMinimumFrequencyEstimatorName("GlobalMinimumFrequencyEstimator");
+static constexpr char const * const         GrainedFrequencyEstimatorName("GrainedFrequencyEstimator");
+
+using GrainedFrequencyEstimator = GrainEstimatorImpl<std::string, FrequencyEstimator<>>;
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         GlobalMinimumFrequencyAnnotation
+///  \brief         This is an annotation class which holds the frequency
+///                 of TimeSeries
+///
+class GlobalMinimumFrequencyAnnotation{
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using FrequencyType                     = std::chrono::system_clock::duration;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Data
+    // |
+    // ----------------------------------------------------------------------
+    FrequencyType                           Value;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    GlobalMinimumFrequencyAnnotation(FrequencyType value);
+    ~GlobalMinimumFrequencyAnnotation(void) = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(GlobalMinimumFrequencyAnnotation);
+};
+
+namespace Details {
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         GlobalMinimumFrequencyTrainingOnlyPolicy
+///  \brief         `GlobalMinimumFrequencyEstimator` implementation details.
+///
+template <typename EstimatorT>

```

Comment:
I don't think that this needs to be template, does it?

Reply:
Implementation has changed to be like IndexMapEstimator. EstimatorT is needed to retrieve annotation.